pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/MetadataStoreFactoryImpl.java

```diff
@@ -45,19 +63,39 @@ public static MetadataStoreExtended createExtended(String metadataURL, MetadataS
     private static MetadataStore newInstance(String metadataURL, MetadataStoreConfig metadataStoreConfig,
                                              boolean enableSessionWatcher)
             throws MetadataStoreException {
+        MetadataStoreProvider provider = findProvider(metadataURL);
```

Comment:
I suggest we should load providers here, and then find the provider by giving the metadataURL.